third_party/move/move-vm/runtime/src/storage/publishing.rs

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    ambassador_impl_ModuleStorage, module_linker_error, IntoUnsyncModuleStorage, Module,
+    ModuleBytesStorage, ModuleStorage, RuntimeEnvironment, UnsyncModuleStorage,
+};
+use ambassador::Delegate;
+use bytes::Bytes;
+use move_binary_format::{
+    access::ModuleAccess,
+    compatibility::Compatibility,
+    errors::{verification_error, Location, PartialVMError, VMResult},
+    normalized, CompiledModule, IndexKind,
+};
+use move_core_types::{
+    account_address::AccountAddress,
+    identifier::{IdentStr, Identifier},
+    metadata::Metadata,
+    vm_status::StatusCode,
+};
+use std::{
+    collections::{btree_map, BTreeMap},
+    sync::Arc,
+};
+
+/// An implementation of [ModuleBytesStorage] that stores temporary changes. If used by
+/// [ModuleStorage], the most recent version of a module will be fetched.
+struct TemporaryModuleBytesStorage<'m, M> {
```

Comment:
Call this `ModulePublishingStorage` or sth to indicate that it is tightly related with module publishign logic?